### DISCUSSION TITLE
KOReader integration into the kindle

### DISCUSSION THREAD

**Discussion by notmarek** [koreader/koreader#12832] [Show and tell]
I have implemented an Epub extractor for the kindle's scanner binary and a helper binary to handle the files and launch koreader, everything works great so far.
I havent done series grouping for the epubs yet but reading data gets synced back to the ui from koreader.
You can find the code at https://github.com/notmarek/jbu
I would be glad for any suggestions of what to add/improve.

https://github.com/user-attachments/assets/2b645a16-0ab2-487d-a82f-c68b021e9fdb

**Comment by Frenzie** [comment]
Looks pretty cool. ^_^

**Reply by shamanNS** [reply]
Bye bye File Manager, never liked you 😂😂

**Comment by mergen3107** [comment]
> I would be glad for any suggestions of what to add/improve.

Is it possible to do for fb2.zip files too?

and pdf/djvu? :D 

P.S. Also, how did you set up remote controls of your Kindle from PC? 
<kbd>¯\(o.O)/¯</kbd>

**Comment by notmarek** [comment]
Opening is easy and doable for any file type with a bit of appreg.db magic,
for indexing i will need to write specific extractors for the file type.
Syncing can be done in multiple ways, right now i have it implemented
inside the extractor and the file is rescanned on koreader exit - for
filetypes that are already indexed by the stock kindle software (pdf, mobi
etc.) It would need to be implemented inside my koreader_helper binary.

**Comment by Commodore64user** [comment]
Would it be compatible with older FW versions and/or devices?

**Reply by notmarek** [reply]
I've only tried it on 5.17.1 and @shamanNS on 5.16.2.1.1

**Reply by shamanNS** [reply]
I could maaaybe test it on a PW1 with fw 5.6.1.1? 😂
But you would probably need to compile the binaries for "kindle touch & pw1" toolchain target?

**Reply by Frenzie** [reply]
There's a chance that such a non-hf build might Just Work™ even if it's not exactly right. I don't know what's the worst that could happen if it doesn't though. ;-)

**Reply by shamanNS** [reply]
> There's a chance that such a non-hf build might Just Work™ even if it's not exactly right. I don't know what's the worst that could happen if it doesn't though. ;-)

Marek has already compiled this for soft-float (for pw2+ target; 5.16.2.1.1 is the last soft float) but Kindles of "K5 family" have an ABI? cutoff that looks like "pw2 or newer" or "touch and pw1". Even KOReader itself is released like that, separate zip for pw2+ and separate for 2 older models on v5.x firmware branch.

**Reply by Frenzie** [reply]
Yes, that's what I'm saying. It still might just work. It's obviously not recommended.

Anyway, I can probably build you one on the oldest TC in an hour or so, if it suffices to pass `-DCMAKE_TOOLCHAIN_FILE`.

**Reply by Commodore64user** [reply]
here I was thinking more along the lines of FW 3.4.3 or 4.1.4 :)

**Reply by Frenzie** [reply]
@shamanNS Case in point, see https://github.com/koreader/koreader/issues/12778#issuecomment-2525551718 ;-)

**Comment by notmarek** [comment]
Yes, that's all you need, I'm not sure if latest stuff i pushed to master
works tho😅

**Reply by Frenzie** [reply]
~~Yeah, it doesn't seem to work.~~

Edit: my bad, my toolchain file there had some things in it that don't really fly for this purpose. :-D

Edit 2: doesn't seem to have been the problem though, so… since I don't have a Kindle and all I'll leave it all to you guys. ;-)
